lib/rack/attack.rb

```diff
@@ -84,9 +84,15 @@ def clear!
 
     attr_reader :configuration
 
-    def initialize(app)
+    def initialize(app, &block)
       @app = app
-      @configuration = self.class.configuration
+      @configuration =
+        if block_given?
+          configuration = Configuration.new
+          configuration.instance_exec(&block)
```

Comment:
You need to make the final statement in this branch `configuration`, otherwise `@configuration` gets set to the return value of the last statement in the block, which in my case was an instance of `Throttle`.

Otherwise, this seems to work nicely, I've monkey patched this change into a project I'm working on it does the job wonderfully.

Reply:
Thanks @jellybob for the code review.

Ideally this problem would reveal in any new test case covering the new block argument.